Library/DiscUtils.Vhdx/LogEntry.cs

```diff
@@ -121,7 +121,7 @@ public static bool TryRead(Stream logStream, out LogEntry entry)
             byte[] logEntryBuffer = new byte[header.EntryLength];
             Array.Copy(sectorBuffer, logEntryBuffer, LogSectorSize);
 
-            StreamUtilities.ReadFully(logStream, logEntryBuffer, LogSectorSize, logEntryBuffer.Length - LogSectorSize);
+            StreamUtilities.ReadExact(logStream, logEntryBuffer, LogSectorSize, logEntryBuffer.Length - LogSectorSize);
```

Comment:
Length handling is taken care of on line 119, returning false if not enough data. However, I missed line 103. It throws on short read, and should return false. Before, it would have returned false on a signature check at line 110.